src/main/java/ch/wisv/areafiftylan/products/service/MolliePaymentService.java

```diff
@@ -42,30 +42,27 @@
 
     private final OrderRepository orderRepository;
 
-    private final Client mollie;
-
     @Value("${a5l.molliekey:null}")
     String apiKey;
 
     @Value("${a5l.paymentReturnUrl}")
     String returnUrl;
 
     @Autowired
-    public MolliePaymentService(OrderRepository orderRepository, @Value("${a5l.molliekey:null}") String apiKey) {
+    public MolliePaymentService(OrderRepository orderRepository) {
         this.orderRepository = orderRepository;
-        this.mollie = new ClientBuilder().withApiKey(apiKey).build();
     }
 
     @Override
     public String registerOrder(Order order) {
+        Client mollie =  new ClientBuilder().withApiKey(apiKey).build();
```

Comment:
This creates a new instance of the Mollie client for each request. Just to rule out any threading issues, as there's a significant delay in communication with Mollie.